cmd/ami-check/main.go

```diff
@@ -0,0 +1,198 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"strconv"
+	"time"
+
+	awsutil "github.com/Comcast/kuberhealthy/pkg/aws"
+	kh "github.com/Comcast/kuberhealthy/pkg/checks/external/checkClient"
+	"github.com/aws/aws-sdk-go/aws/session"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/client-go/kubernetes"
+)
+
+var (
+	// AWS region variables.
+	awsRegionEnv = os.Getenv("AWS_REGION")
+	awsRegion    string
+
+	// AWS S3 Bucket variables.
+	awsS3BucketNameEnv = os.Getenv("AWS_S3_BUCKET_NAME")
+	awsS3BucketName    string
+
+	// Name of the cluster for Kops state store.
+	clusterNameEnv = os.Getenv("CLUSTER_FQDN")
+	clusterName    string
+
+	signalChan chan os.Signal
+
+	debugEnv = os.Getenv("DEBUG")
+	debug    bool
+
+	ctx context.Context
+
+	// AWS session.
+	awsSess *session.Session
+
+	// K8s client.
+	k8sClient *kubernetes.Clientset
+)
+
+const (
+	// Matching strings for kops bucket operations.
+	regexpAWSRegion                 = `^[\w]{2}[-][\w]{4,9}[-][\d]$`
+	regexpKopsStateStoreS3ObjectKey = `/instancegroup/`
+
+	// Default AWS region.
+	defaultAWSRegion = "us-east-1"
+
+	// Default AWS S3 bucket name.
+	defaultAWSS3BucketName = "kops-state-store"
+
+	// Default cluster FQDN
+	defaultClusterName = "cluster-fqdn"
+)
+
+func init() {
+
+	// Parse AWS_REGION environment variable.
+	awsRegion = defaultAWSRegion
+	if len(awsRegionEnv) != 0 {
+		awsRegion = awsRegionEnv
+		ok, err := regexp.Match(regexpAWSRegion, []byte(awsRegion))
+		if err != nil {
+			log.Fatalln("Failed to parse AWS_REGION:", err.Error())
+		}
+		if !ok {
+			log.Fatalln("Given AWS_REGION does not match AWS Region format.")
+		}
+	}
+
+	// Parse AWS_S3_BUCKET_NAME environment variable.
+	awsS3BucketName = defaultAWSS3BucketName
+	if len(awsS3BucketNameEnv) != 0 {
+		awsS3BucketName = awsS3BucketNameEnv
+		if len(awsS3BucketName) == 0 {
```

Comment:
Not sure why theres a double check on whether `awsS3BucketNameEnv` is empty? Not sure if its necessary after you've set `awsS3BucketName = awsS3BucketNameEnv`

Reply:
took this out!